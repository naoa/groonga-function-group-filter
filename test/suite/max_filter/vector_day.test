plugin_register functions/group_filter
table_create docs TABLE_HASH_KEY ShortText
table_create dates TABLE_HASH_KEY Time
column_create docs date COLUMN_VECTOR dates
load --table docs
[
{"_key": "JP2001213456", "date": ["2019-01-01 00:00:00"]},
{"_key": "JP2001213457", "date": ["2019-03-01 00:00:00", "2019-03-02 00:00:00"]},
{"_key": "JP2001213458", "date": ["2019-03-02 00:00:00", "2019-03-03 00:00:00"]},
{"_key": "JP2001213459", "date": ["2019-03-03 00:00:00"]}
]

column_create dates dates COLUMN_INDEX docs date

select docs \
  --filter '_id > 0 && max_filter("date", 1)' \
  --output_columns '_id,_score,_key,date' --command_version 2
