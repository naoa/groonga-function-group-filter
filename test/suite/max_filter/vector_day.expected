plugin_register functions/group_filter
[[0,0.0,0.0],true]
table_create docs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create dates TABLE_HASH_KEY Time
[[0,0.0,0.0],true]
column_create docs date COLUMN_VECTOR dates
[[0,0.0,0.0],true]
load --table docs
[
{"_key": "JP2001213456", "date": ["2019-01-01 00:00:00"]},
{"_key": "JP2001213457", "date": ["2019-03-01 00:00:00", "2019-03-02 00:00:00"]},
{"_key": "JP2001213458", "date": ["2019-03-02 00:00:00", "2019-03-03 00:00:00"]},
{"_key": "JP2001213459", "date": ["2019-03-03 00:00:00"]}
]
[[0,0.0,0.0],4]
column_create dates dates COLUMN_INDEX docs date
[[0,0.0,0.0],true]
select docs   --filter '_id > 0 && max_filter("date", 1)'   --output_columns '_id,_score,_key,date' --command_version 2
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Float"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "date",
          "dates"
        ]
      ],
      [
        3,
        1.0,
        "JP2001213458",
        [
          1551484800.0,
          1551571200.0
        ]
      ],
      [
        4,
        1.0,
        "JP2001213459",
        [
          1551571200.0
        ]
      ]
    ]
  ]
]
